### PR TITLE
Fix/motor issues

### LINKED ISSUE #573: Building Motor doesn't export Init() method
#### Describe the bug

The init method is no longer binded to Motor target platforms (obj-c, java), preventing any actions on the Motor node.

### PR DESCRIPTION
**Pull request by prdlk**
## Changes

Fixes #573 #574 #572
Connects

 Mention [stepsize] in a comment if you'd like to report some technical debt. See examples here.

### CODE REVIEW

**Review by ntindle — APPROVED** [review]
I’d still use the constants for messages but that’s just me

**Comment by prdlk** [comment]
> I’d still use the constants for messages but that’s just me

Can go direct to logging on interface on the client during the loading process in UI. This way there is rationale behind long lengths of time for account creation for now. @SCKelemen

File: pkg/motor/create_account.go [hunk 1]
```diff
@@ -9,29 +9,26 @@ import (
...
 	"github.com/sonr-io/sonr/pkg/vault"
-	ct "github.com/sonr-io/sonr/third_party/types/common"
 	mt "github.com/sonr-io/sonr/third_party/types/motor/api/v1"
 	rt "github.com/sonr-io/sonr/x/registry/types"
 )
 
 func (mtr *motorNodeImpl) CreateAccount(request mt.CreateAccountRequest) (mt.CreateAccountResponse, error) {
 	// create motor
-	mtr.callback.OnMotorEvent(ct.MotorCallbackMessage_MTR_INIT, false)
+	mtr.callback.OnMotorEvent("Initializing motor", false)
```
Comment (ma-sonr) [inline pkg/motor/create_account.go:18]: References to the value represented by each string should be made through some kind of strongly typed value / constant

File: pkg/motor/create_account.go [hunk 2]
```diff
@@ -9,29 +9,26 @@ import (
...
 	if err := initMotor(mtr); err != nil {
 		return mt.CreateAccountResponse{}, fmt.Errorf("initialize motor: %s", err)
 	}
-	mtr.callback.OnMotorEvent(ct.MotorCallbackMessage_MTR_INIT, true)
 
 	// Request from Faucet
-	mtr.callback.OnMotorEvent(ct.MotorCallbackMessage_MTR_FAUCET_RECEIVED, false)
+	mtr.callback.OnMotorEvent("Requesting Airdrop for initial balance", false)
```
Comment (ma-sonr) [inline pkg/motor/create_account.go:24]: References to the value represented by each string should be made through some kind of strongly typed value / constant

File: pkg/motor/create_account.go [hunk 3]
```diff
@@ -9,29 +9,26 @@ import (
...
 	err := mtr.Cosmos.RequestFaucet(mtr.Address)
 	if err != nil {
 		return mt.CreateAccountResponse{}, fmt.Errorf("request from faucet: %s", err)
 	}
-	mtr.callback.OnMotorEvent(ct.MotorCallbackMessage_MTR_FAUCET_RECEIVED, true)
 
 	// Create Initial Shards
-	mtr.callback.OnMotorEvent(ct.MotorCallbackMessage_MTR_SHARDS_CREATED, false)
+	mtr.callback.OnMotorEvent("Creating shards for MPC", false)
```
Comment (ma-sonr) [inline pkg/motor/create_account.go:31]: References to the value represented by each string should be made through some kind of strongly typed value / constant

File: pkg/motor/create_account.go [hunk 4]
```diff
@@ -50,13 +46,14 @@ func (mtr *motorNodeImpl) CreateAccount(request mt.CreateAccountRequest) (mt.Cre
 	mtr.DIDDocument = doc
 
 	// create Vault shards to make sure this works before creating WhoIs
-	mtr.callback.OnMotorEvent(ct.MotorCallbackMessage_MTR_VAULT_CREATED, false)
+	mtr.callback.OnMotorEvent("Registering new DIDDocument for account", false)
```
Comment (ma-sonr) [inline pkg/motor/create_account.go:49]: References to the value represented by each string should be made through some kind of strongly typed value / constant

File: pkg/motor/create_account.go [hunk 5]
```diff
@@ -50,13 +46,14 @@ func (mtr *motorNodeImpl) CreateAccount(request mt.CreateAccountRequest) (mt.Cre
...
 	vc := vault.New()
 	if _, err := createWhoIs(mtr); err != nil {
 		return mt.CreateAccountResponse{}, fmt.Errorf("create account: %s", err)
 	}
 
 	// ecnrypt dscShard with DSC
+	mtr.callback.OnMotorEvent("Encrypting shards for Vault", false)
```
Comment (ma-sonr) [inline pkg/motor/create_account.go:56]: References to the value represented by each string should be made through some kind of strongly typed value / constant

File: pkg/motor/create_account.go [hunk 6]
```diff
@@ -75,6 +72,7 @@ func (mtr *motorNodeImpl) CreateAccount(request mt.CreateAccountRequest) (mt.Cre
 	}
 
 	// create vault
+	mtr.callback.OnMotorEvent("Setting up Account Vault", false)
```
Comment (ma-sonr) [inline pkg/motor/create_account.go:75]: References to the value represented by each string should be made through some kind of strongly typed value / constant

File: pkg/motor/create_account.go [hunk 7]
```diff
@@ -86,17 +84,17 @@ func (mtr *motorNodeImpl) CreateAccount(request mt.CreateAccountRequest) (mt.Cre
 	if err != nil {
 		return mt.CreateAccountResponse{}, fmt.Errorf("setup vault: %s", err)
 	}
-	mtr.callback.OnMotorEvent(ct.MotorCallbackMessage_MTR_VAULT_CREATED, true)
 
 	// update DID Document
+	mtr.callback.OnMotorEvent("Updating DIDDocument for Account", false)
```
Comment (ma-sonr) [inline pkg/motor/create_account.go:89]: References to the value represented by each string should be made through some kind of strongly typed value / constant

File: pkg/motor/create_account.go [hunk 8]
```diff
@@ -86,17 +84,17 @@ func (mtr *motorNodeImpl) CreateAccount(request mt.CreateAccountRequest) (mt.Cre
...
 	mtr.DIDDocument.AddService(vaultService)
 
 	// update whois
 	resp, err := updateWhoIs(mtr)
 	if err != nil {
 		return mt.CreateAccountResponse{}, fmt.Errorf("update WhoIs: %s", err)
 	}
-
+	mtr.callback.OnMotorEvent("Account registered successfully!", true)
```
Comment (ma-sonr) [inline pkg/motor/create_account.go:97]: References to the value represented by each string should be made through some kind of strongly typed value / constant

File: pkg/motor/login.go [hunk 9]
```diff
@@ -22,12 +21,13 @@ func (mtr *motorNodeImpl) Login(request mt.LoginRequest) (mt.LoginResponse, erro
 	mtr.Address = request.Did
 
 	// fetch vault shards
-	mtr.callback.OnMotorEvent(ct.MotorCallbackMessage_MTR_LOGGED_IN, false)
+	mtr.callback.OnMotorEvent("Fetching shards from vault", false)
```
Comment (ma-sonr) [inline pkg/motor/login.go:24]: References to the value represented by each string should be made through some kind of strongly typed value / constant

File: pkg/motor/login.go [hunk 10]
```diff
@@ -22,12 +21,13 @@ func (mtr *motorNodeImpl) Login(request mt.LoginRequest) (mt.LoginResponse, erro
...
 	shards, err := vault.New().GetVaultShards(request.Did)
 	if err != nil {
 		return mt.LoginResponse{}, fmt.Errorf("error getting vault shards: %s", err)
 	}
 
+	mtr.callback.OnMotorEvent("Reconstructing wallet", false)
```
Comment (ma-sonr) [inline pkg/motor/login.go:30]: References to the value represented by each string should be made through some kind of strongly typed value / constant

File: pkg/motor/login.go [hunk 11]
```diff
@@ -39,6 +39,7 @@ func (mtr *motorNodeImpl) Login(request mt.LoginRequest) (mt.LoginResponse, erro
 	}
 
 	// fetch DID document from chain
+	mtr.callback.OnMotorEvent("Verifying with Blockchain", false)
```
Comment (ma-sonr) [inline pkg/motor/login.go:42]: References to the value represented by each string should be made through some kind of strongly typed value / constant

File: pkg/motor/login.go [hunk 12]
```diff
@@ -56,8 +57,7 @@ func (mtr *motorNodeImpl) Login(request mt.LoginRequest) (mt.LoginResponse, erro
 	mtr.sharedShard = shards.PskShard
 	mtr.recoveryShard = shards.RecoveryShard
 	mtr.unusedShards = destructureShards(shards.ShardBank)
-	mtr.callback.OnMotorEvent(ct.MotorCallbackMessage_MTR_LOGGED_IN, true)
-
+	mtr.callback.OnMotorEvent("Logged into account successfully!", true)
```
Comment (ma-sonr) [inline pkg/motor/login.go:60]: References to the value represented by each string should be made through some kind of strongly typed value / constant

**Review by ma-sonr — CHANGES_REQUESTED** [review]

**Review by ma-sonr — APPROVED** [review]

**Review by ma-sonr — APPROVED** [review]

**Pull request merged**